presto-main/src/main/java/com/facebook/presto/type/IntervalYearMonthOperators.java

```diff
@@ -170,6 +172,13 @@ public static long hashCode(@SqlType(StandardTypes.INTERVAL_YEAR_TO_MONTH) long
         return AbstractIntType.hash((int) value);
     }
 
+    @ScalarOperator(XX_HASH_64)
+    @SqlType(StandardTypes.BIGINT)
+    public static long xxHash64(@SqlType(StandardTypes.INTERVAL_YEAR_TO_MONTH) long value)
+    {
+        return XxHash64.hash((int) value);
```

Comment:
Casting to `int` seems to be extra here.

Reply:
`IntervalYearMonth` type is a 4 bytes `AbstractIntType`. It seems valid to me to cast it to `int`.  We also did this in the `HASH_CODE` operator.  It may be redundant though since we should never have overflowed values.

Reply:
Deleted my previous comment. 

I realized that the cast in `hashcode()` implemenatation is for the `AbstractIntType.hash()`. So there is no point here to cast it to `int`. I will remove it.